Auto update failing for tere

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the tere package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="bash">
INFO: Updating tere [Current version: 1.4.0]
INFO: package being updated to 1.5.0.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  285k    0  285k    0     0   412k      0 --:--:-- --:--:-- --:--:--  412k

You are about to commit these changes:

--------------------
diff --git a/packages/tere/build.sh b/packages/tere/build.sh
index a9fdf123e..6fe2ab472 100644
--- a/packages/tere/build.sh
+++ b/packages/tere/build.sh
@@ -3,9 +3,9 @@ TERMUX_PKG_DESCRIPTION="Terminal file explorer written in rust"
 TERMUX_PKG_LICENSE="EUPL-1.2"
 TERMUX_PKG_LICENSE_FILE="LICENSE"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="1.4.0"
+TERMUX_PKG_VERSION="1.5.0"
 TERMUX_PKG_SRCURL=https://github.com/mgunyho/tere/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=84eeafc346ee2207bcfb0a9e29a6a4e7748817741a0f7245a204d16da0ef651f
+TERMUX_PKG_SHA256=84195f45b738fb7c805d7b348185658d0dc58aa26e7f92fcad9578bc7bd694bf
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_AUTO_UPDATE=true
 
--------------------

bump(main/tere): 1.5.0


Not committing to Git!

INFO: Trying to build package.
INFO: Package tere exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building tere for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13134  100 13134    0     0  50390      0 --:--:-- --:--:-- --:--:-- 50515
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1554      0 --:--:-- --:--:-- --:--:--  1557
gpg: Signature made Mon 24 Jul 2023 12:16:30 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13320  100 13320    0     0  24656      0 --:--:-- --:--:-- --:--:-- 24666
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1595      0 --:--:-- --:--:-- --:--:--  1592
100   833  100   833    0     0   1594      0 --:--:-- --:--:-- --:--:--  1592
gpg: Signature made Fri 21 Jul 2023 01:56:11 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13120  100 13120    0     0  25275      0 --:--:-- --:--:-- --:--:-- 25328
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1546      0 --:--:-- --:--:-- --:--:--  1545
100   833  100   833    0     0   1546      0 --:--:-- --:--:-- --:--:--  1545
gpg: Signature made Mon 24 Jul 2023 12:58:34 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://github.com/mgunyho/tere/archive/refs/tags/v1.5.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  285k    0  285k    0     0   448k      0 --:--:-- --:--:-- --:--:--  834k
info: downloading installer
info: profile set to 'default'
info: default host triple is x86_64-unknown-linux-gnu
info: syncing channel updates for '1.70.0-x86_64-unknown-linux-gnu'
info: latest update on 2023-06-01, rust version 1.70.0 (90c541806 2023-05-31)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustfmt'

info: default toolchain set to '1.70.0-x86_64-unknown-linux-gnu'
  1.70.0-x86_64-unknown-linux-gnu installed - rustc 1.70.0 (90c541806 2023-05-31)


Rust is installed now. Great!

To get started you may need to restart your current shell.
This would reload your PATH environment variable to include
Cargo's bin directory ($HOME/.cargo/bin).

To configure your current shell, run:
source "$HOME/.cargo/env"
info: downloading component 'rust-std' for 'aarch64-linux-android'
info: installing component 'rust-std' for 'aarch64-linux-android'
    Updating crates.io index
 Downloading crates ...
  Downloaded clap_lex v0.3.0
  Downloaded io-lifetimes v0.7.5
  Downloaded errno v0.2.8
  Downloaded dirs-sys v0.3.6
  Downloaded clap_derive v4.0.21
  Downloaded bitflags v1.3.2
  Downloaded terminal_size v0.2.2
  Downloaded signal-hook-mio v0.2.3
  Downloaded itoa v1.0.1
  Downloaded version_check v0.9.4
  Downloaded strum v0.24.1
  Downloaded autocfg v1.1.0
  Downloaded strsim v0.10.0
  Downloaded smallvec v1.8.0
  Downloaded unicode-width v0.1.9
  Downloaded proc-macro2 v1.0.43
  Downloaded unicode-ident v1.0.3
  Downloaded unicode-linebreak v0.1.2
  Downloaded unicode-segmentation v1.9.0
  Downloaded serde v1.0.144
  Downloaded serde_json v1.0.85
  Downloaded clap v4.0.32
  Downloaded mio v0.8.4
  Downloaded crossterm v0.24.0
  Downloaded aho-corasick v0.7.18
  Downloaded signal-hook v0.3.13
  Downloaded syn v1.0.99
  Downloaded regex-syntax v0.6.27
  Downloaded regex v1.6.0
  Downloaded rustix v0.35.9
  Downloaded memchr v2.4.1
  Downloaded textwrap v0.15.0
  Downloaded serde_derive v1.0.144
  Downloaded parking_lot v0.12.1
  Downloaded once_cell v1.13.1
  Downloaded linux-raw-sys v0.0.46
  Downloaded crokey v0.5.0
  Downloaded libc v0.2.146
  Downloaded parking_lot_core v0.9.3
  Downloaded log v0.4.14
  Downloaded strum_macros v0.24.3
  Downloaded signal-hook-registry v1.4.0
  Downloaded ryu v1.0.9
  Downloaded rustversion v1.0.9
  Downloaded quote v1.0.21
  Downloaded proc-macro-error v1.0.4
  Downloaded os_str_bytes v6.0.0
  Downloaded lock_api v0.4.7
  Downloaded smawk v0.3.1
  Downloaded scopeguard v1.1.0
  Downloaded heck v0.4.0
  Downloaded dirs v4.0.0
  Downloaded crokey-proc_macros v0.4.0
  Downloaded cfg-if v1.0.0
  Downloaded proc-macro-error-attr v1.0.4
   Compiling libc v0.2.146
   Compiling proc-macro2 v1.0.43
   Compiling quote v1.0.21
   Compiling unicode-ident v1.0.3
   Compiling syn v1.0.99
   Compiling memchr v2.4.1
   Compiling version_check v0.9.4
   Compiling cfg-if v1.0.0
   Compiling autocfg v1.1.0
   Compiling log v0.4.14
   Compiling lock_api v0.4.7
   Compiling aho-corasick v0.7.18
   Compiling proc-macro-error-attr v1.0.4
   Compiling signal-hook v0.3.13
   Compiling bitflags v1.3.2
   Compiling serde_derive v1.0.144
   Compiling io-lifetimes v0.7.5
   Compiling parking_lot_core v0.9.3
   Compiling regex-syntax v0.6.27
   Compiling regex v1.6.0
   Compiling proc-macro-error v1.0.4
   Compiling signal-hook-registry v1.4.0
   Compiling rustix v0.35.9
   Compiling scopeguard v1.1.0
   Compiling rustversion v1.0.9
   Compiling heck v0.4.0
   Compiling smallvec v1.8.0
   Compiling serde v1.0.144
   Compiling unicode-linebreak v0.1.2
   Compiling mio v0.8.4
   Compiling errno v0.2.8
   Compiling linux-raw-sys v0.0.46
error[E0308]: arguments to this function are incorrect
    --> /home/builder/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.35.9/src/backend/libc/fs/makedev.rs:15:14
     |
15   |     unsafe { c::makedev(maj as i32, min as i32) }
     |              ^^^^^^^^^^ ----------  ---------- expected `u32`, found `i32`
     |                         |
     |                         expected `u32`, found `i32`
     |
note: function defined here
    --> /home/builder/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.146/src/unix/linux_like/android/mod.rs:3110:20
     |
3110 |     pub {const} fn makedev(ma: ::c_uint, mi: ::c_uint) -> ::dev_t {
     |                    ^^^^^^^
help: you can convert an `i32` to a `u32` and panic if the converted value doesn't fit
     |
15   |     unsafe { c::makedev((maj as i32).try_into().unwrap(), min as i32) }
     |                         +          +++++++++++++++++++++
help: you can convert an `i32` to a `u32` and panic if the converted value doesn't fit
     |
15   |     unsafe { c::makedev(maj as i32, (min as i32).try_into().unwrap()) }
     |                                     +          +++++++++++++++++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `rustix` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2023-07-24 12:22:17 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/5644588149">5644588149</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/tere/0001-rustix-fix-libc-removing-unsafe-on-makedev.diff
+++ b/packages/tere/0001-rustix-fix-libc-removing-unsafe-on-makedev.diff
@@ -1,0 +1,30 @@
+diff --git a/src/backend/libc/fs/makedev.rs b/src/backend/libc/fs/makedev.rs
+index d9089e7f..08ecd872 100644
+--- a/src/backend/libc/fs/makedev.rs
++++ b/src/backend/libc/fs/makedev.rs
+@@ -5,14 +5,14 @@ use crate::fs::Dev;
+ #[cfg(not(any(target_os = "android", target_os = "emscripten")))]
+ #[inline]
+ pub(crate) fn makedev(maj: u32, min: u32) -> Dev {
+-    unsafe { c::makedev(maj, min) }
++    c::makedev(maj, min)
+ }
+ 
+ #[cfg(all(target_os = "android", not(target_pointer_width = "32")))]
+ #[inline]
+ pub(crate) fn makedev(maj: u32, min: u32) -> Dev {
+     // Android's `makedev` oddly has signed argument types.
+-    unsafe { c::makedev(maj as i32, min as i32) }
++    c::makedev(maj, min)
+ }
+ 
+ #[cfg(all(target_os = "android", target_pointer_width = "32"))]
+@@ -30,7 +30,7 @@ pub(crate) fn makedev(maj: u32, min: u32) -> Dev {
+ #[inline]
+ pub(crate) fn makedev(maj: u32, min: u32) -> Dev {
+     // Emscripten's `makedev` has a 32-bit return value.
+-    Dev::from(unsafe { c::makedev(maj, min) })
++    Dev::from(c::makedev(maj, min))
+ }
+ 
+ #[cfg(not(any(target_os = "android", target_os = "emscripten")))]

--- a/packages/tere/build.sh
+++ b/packages/tere/build.sh
@@ -3,11 +3,27 @@ TERMUX_PKG_DESCRIPTION="Terminal file explorer written in rust"
 TERMUX_PKG_LICENSE="EUPL-1.2"
 TERMUX_PKG_LICENSE_FILE="LICENSE"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="1.4.0"
+TERMUX_PKG_VERSION="1.5.0"
 TERMUX_PKG_SRCURL=https://github.com/mgunyho/tere/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=84eeafc346ee2207bcfb0a9e29a6a4e7748817741a0f7245a204d16da0ef651f
+TERMUX_PKG_SHA256=84195f45b738fb7c805d7b348185658d0dc58aa26e7f92fcad9578bc7bd694bf
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_AUTO_UPDATE=true
+
+termux_step_pre_configure() {
+	termux_setup_rust
+
+	export CFLAGS="${TARGET_CFLAGS}"
+
+	: "${CARGO_HOME:=$HOME/.cargo}"
+	export CARGO_HOME
+
+	rm -rf $CARGO_HOME/registry/src/*/rustix-*
+	cargo fetch --target "${CARGO_TARGET_NAME}"
+
+	for d in $CARGO_HOME/registry/src/*/rustix-*; do
+		patch --silent -p1 -d ${d} < $TERMUX_PKG_BUILDER_DIR/0001-rustix-fix-libc-removing-unsafe-on-makedev.diff || :
+	done
+}
 
 termux_step_make() {
 	termux_setup_rust

